### ISSUE TITLE
Helm Release History not cleared when bundle deleted

### ISSUE THREAD

**Issue by boxboatmatt** [rancher/fleet#233]
When a bundle deployment is deleted the helm release history is not being cleared

Addressed by https://github.com/rancher/fleet/pull/231

`pkg/helmdeployer/deployer.go` is never calling `deleteHistory`.

**Comment by boxboatmatt** [comment]
@StrongMonkey I think I saw this issue occurring on my cluster when I updated my `fleet.yaml` to use a different release name... As you noted in #232, a helm uninstall will remove all of the history. I wrote this bug mostly on the basis of the code having a logic bug see https://github.com/rancher/fleet/pull/231.

**Comment by boxboatmatt** [comment]
I have a `GitRepo` configured to scan `/test`, I put a helm chart in `/test/fleet.yaml`. Fleet installs the bundle properly (interestingly I get 2 release secrets created - `.v1` and `.v2`). If I delete the bundle from the Rancher CD UI, the helm release does not get deleted  or uninstalled from the target cluster; however the `bundledeployment` does get deleted from the rancher cluster.

**Comment by boxboatmatt** [comment]
Another observation - though this may be a separate bug. If you update the `fleet.yaml` `helm.releaseName` then the old helm release is not uninstalled.

**Comment by StrongMonkey** [comment]
At this time I don't think we handle the case where you modify the release name. Does the behavior still happen when you don't modify the release name in fleet.yaml?

**Comment by boxboatmatt** [comment]
Yes... if you delete the bundle from rancher the release is not uninstalled from the target.

**Comment by StrongMonkey** [comment]
I still can't reproduce this issue.
The steps I took:
1. Create a git repo https://github.com/rancher/fleet-examples.git and path `multi-cluster/helm-external`.
2. Bundle and helm app are created.
3. Delete git Repo.
4. Releases are deleted.

@boxboatmatt Can you elobarate on the steps and what your manifest looks like?

**Comment by StrongMonkey** [comment]
Basically we found a bug that if you specify releaseName in fleet.yaml, we don't clean up the release properly. For example: 
```
namespace: test
helm:
  # The release name to use. If empty a generated release name will be used
  releaseName: mongo
  chart: "mongodb"
  repo: "https://charts.bitnami.com/bitnami"
  version: ""
  force: false
  timeoutSeconds: 0
  values:
    auth:
      enabled: false
```
This is fixed by https://github.com/rancher/fleet/pull/294. @jiaqiluo Basically you can reproduce this by creating a git repo with above fleet.yaml and try to delete the gitrepo. The actual resource will not be deleted. v0.3.4-rc7 should fix this.

**Comment by jiaqiluo** [comment]
The bug is reproduced on Rancher v2.5.5 

The key here is to give the gitRepo a name that is different from the `releaseName` in the `fleet.yaml`.
For example, I named the gitRepo `test-1` while the `releaseName` is [`guestbook`](https://github.com/rancher/fleet-examples/blob/a4364e622733fb84032d3688146d8cfe9dbd986e/single-cluster/helm/fleet.yaml#L10)

 
<img width="1102" alt="Screen Shot 2021-02-26 at 3 46 19 PM" src="https://user-images.githubusercontent.com/6218999/109363509-feefb480-7849-11eb-9dcd-c41625057d5b.png">

After the gitRepo is active, we can find the following helm release history as secrets in the namespace where the app is installed.
![image](https://user-images.githubusercontent.com/6218999/109363731-7ae9fc80-784a-11eb-83ed-bb412750ef39.png)

Now delete the gitRepo, those helm release histories are left in the namespace.

**Comment by jiaqiluo** [comment]
The bug fix is validated on v2.5-head 1b5aad6 with fleet chart `0.3.400-rc07`

Following the same steps as my previous comment describes, now the helm release histories are deleted after the gitReop is deleted.
